opm/simulators/wells/WellState.hpp

```diff
@@ -282,9 +312,37 @@ namespace Opm
 
         WellMapType wellMap_;
 
+        using MPIComm = typename Dune::MPIHelper::MPICommunicator;
+#if DUNE_VERSION_NEWER(DUNE_COMMON, 2, 7)
+        using Communication = Dune::Communication<MPIComm>;
+#else
+        using Communication = Dune::CollectiveCommunication<MPIComm>;
+#endif
+        void gatherVectorsOnRoot(const std::vector< data::Connection >& from_connections,
+                                 std::vector< data::Connection >& to_connections,
+                                 const Communication& comm) const
```

Comment:
If you did Communication as a template argument you could skip the #if parts above. On the other hand, this being concrete means it can be put in a cpp file. I am thinking that perhaps the methods that were moved to the WellGroupHelpers.hpp header due to the comm argument could be put back in the cpp file?